kvbc/include/categorization/kv_blockchain.h

```diff
@@ -90,25 +133,56 @@ class KeyValueBlockchain {
                                 const MerkleUpdatesInfo& updates_info,
                                 storage::rocksdb::NativeWriteBatch&);
 
+  /////////////////////// Updates ///////////////////////
+
   // Update per category
   MerkleUpdatesInfo handleCategoryUpdates(BlockId block_id,
                                           const std::string& category_id,
                                           MerkleUpdatesData&& updates,
-                                          concord::storage::rocksdb::NativeWriteBatch& write_batch);
+                                          concord::storage::rocksdb::NativeWriteBatch& write_batch,
+                                          categorization::RawBlock& raw_block);
 
   KeyValueUpdatesInfo handleCategoryUpdates(BlockId block_id,
                                             const std::string& category_id,
                                             KeyValueUpdatesData&& updates,
-                                            concord::storage::rocksdb::NativeWriteBatch& write_batch);
+                                            concord::storage::rocksdb::NativeWriteBatch& write_batch,
+                                            categorization::RawBlock& raw_block);
   ImmutableUpdatesInfo handleCategoryUpdates(BlockId block_id,
                                              const std::string& category_id,
                                              ImmutableUpdatesData&& updates,
-                                             concord::storage::rocksdb::NativeWriteBatch& write_batch);
+                                             concord::storage::rocksdb::NativeWriteBatch& write_batch,
+                                             categorization::RawBlock& raw_block);
+
+  /////////////////////// Members ///////////////////////
 
-  // Members
   std::shared_ptr<concord::storage::rocksdb::NativeClient> native_client_;
+  std::map<std::string, std::variant<detail::ImmutableKeyValueCategory, MerkleCategory, KVHashCategory>> categorires_;
+  std::map<std::string, detail::CATEGORY_TYPE> categorires_types_;
   detail::Blockchain block_chain_;
   detail::Blockchain::StateTransfer state_transfer_block_chain_;
+
+  // Holds the last raw block of the last corresponding block that was added.
+  // Used to save construction of a raw block per addBlock method.
+  // E.L - compare this with getRawBlock to see they are equal
+  VersionedRawBlock last_raw_block_;
+
+ public:
+  struct KeyValueBlockchain_tester {
```

Comment:
Nit: Maybe we can make the members we need for testing `public` and just use them. Rationale is that we are in the `detail` namespace anyway, i.e. we care about encapsulation a bit less. Just a suggestion, no need to change.